.buildbot.sh

```diff
@@ -1,8 +1,19 @@
 #!/bin/sh
-#
-# Build script for continuous integration.
 
-PATH=/opt/rust/bin:${PATH}
-LLVM_READOBJ_PATH=/opt/llvm-stackmapv3-tools/build/bin/llvm-readobj
+#Cleanup code should always happen
+cleanup() {
+    rm -rf .rustup/
+    rm -rf .cargo/
+}
 
+# The build script
+RUST_TOOLCHAIN="stable"
+
+set -e 
+
+curl -sSf "https://sh.rustup.rs" | sh -s -- --default-toolchain=$RUST_TOOLCHAIN -y
+export PATH=/opt/llvm-stackmapv3-tools/build/bin/:$HOME/.cargo/bin:$PATH
+export LLVM_READOBJ_PATH=/opt/llvm-stackmapv3-tools/build/bin/llvm-readobj
 cargo test
+
+trap cleanup EXIT
```

Comment:
Does that get called when there's an error too?

Reply:
Yes, `cleanup` gets run from any exit from the script